### PR TITLE
Removes storybook; Updates Text, Button, and Screen components; Starts updating theming.

### PR DESCRIPTION
**Pull request by yulolimum**
## Please verify the following:

- [] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [] `README.md` has been updated with your changes, if relevant

## Describe your PR

❗❗ Note, since this is a progress PR, some stuff isn't complete 100% (including failing tests). Things that I will need to address are marked with `MAVERICKTODO` comment tag.❗❗

This PR completes the following tasks:

- **Storybook:** Removes Storybook and mostly all of references to it. This includes things like configuration files, stories, documentation. The only things that are left to update are the generators and template files but I'll revisit this at a later time once the new component patterns and the storybook replacement feature is established.
- **Theming:** Starts initial work on theming updates. This includes things like consolidation of palette/colors, typography/fonts, implementation of colors levels and default fonts as per the new design system provided by Jenna. There is some work that still needs to be done with documenting the changes but that too I'll come back to at a later time.
- **New Component Patterns:** In the Ignite demo at the company retreat, I showed a few changes that I would love to see make it into Ignite. This includes:
  -  A flat components/screens folder (i.e. removing the need for `component-name/component-name.ts` subfolder). 
  -  Renaming any component files using PascalCase to better visually differentiate the  component files from other files types especially when using VSCode's quick-open feature (p.s. this doesn't stop people from having `ComponentName/sub-files.ts` if needed - this is especially true for screens that might have one-off sub-components). 
  - Combining the component building blocks into a single file (i.e. putting styles, prop type defs, and presets into the tsx file). 
  - Making the components less assumptive about the customizations it requires and thus making them more customizable. This involves adding more props or making existing props less strict as far as what can be passed to the sub-components.
  - An experimental change for styling variable names: changes from `MACRO_CASE` to `$camelCase` with a dollar sign. This makes those variable names less in-your-face and you can now see style code-completion by typing the `$`. This was introduced by @jamonholmgren  and I have used it ever since with good success. 
- **Text & Button Components**: Implements the 2 said components employing the above stylistic changes. 
- **Screen Component**: Refactors the screen component making it more DRY yet more customizable by exposing sub-component's props for customization. This resolves a few open issues that requested more customizability. 
- **New Welcome Screen**: Using the newly built components, the Welcome screen has been updated to reflect the new design.

## Proofs

| Text Component | Button Component | Welcome Screen |
|---|---|---|
| **presets**<br>yulolimum-capture-2022-08-04--01-10-39@2x | **presets**<br>yulolimum-capture-2022-08-04--01-12-16@2x | yulolimum-capture-2022-08-04--01-13-50@2x |
| **size prop**<br>yulolimum-capture-2022-08-04--01-10-07@2x | **content customization**<br>yulolimum-capture-2022-08-04--01-12-50@2x |  |
| **weight prop**<br>yulolimum-capture-2022-08-04--01-09-41@2x | **styling**<br>yulolimum-capture-2022-08-04--01-13-24@2x |  |
| **content customization**<br>yulolimum-capture-2022-08-04--01-10-56@2x |  |  |
| **styling**<br>yulolimum-capture-2022-08-04--01-11-25@2x |  |  |
|  |  |  |

### CODE REVIEW

File: boilerplate/README.md [hunk 1]
```diff
@@ -88,7 +80,7 @@ app
 ```
 
 **components**
-This is where your React components will live. Each component will have a directory containing the `.tsx` file, along with a story file, and optionally `.presets`, and `.props` files for larger components. The app will come with some commonly used components like Button.
+MAVERICKTODO: update this documentation
```
Comment (yulolimum) [inline boilerplate/README.md:83]: Im adding this tag wherever I need to revisit and update stuff.

File: boilerplate/app/app.tsx [hunk 2]
```diff
@@ -11,14 +11,14 @@
  */
```
Comment (yulolimum) [inline boilerplate/app/app.tsx:11]: Changes to this file include removing the Storybook wrapper as well as adding font-loading (the old way was a placeholder anyways).

File: boilerplate/app/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
+import React, { ComponentType } from "react"
```
Comment (yulolimum) [inline boilerplate/app/components/Button.tsx:1]: New Button component reflecting all stylistic changes and adding more customization options.

File: boilerplate/app/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    pressableState: PressableStateCallbackType
+  }>
+}
+
+/**
+ * A component that allows users to take actions and make choices.
+ * Wraps the Text component with a Pressable component.
+ *
+ * - [Documentation and Examples](https://github.com/infinitered/ignite/blob/master/docs/Components-Button.md)
+ */
```
Comment (yulolimum) [inline boilerplate/app/components/Button.tsx:68]: Adds updated ts doc comment including a link to the actual markdown file.

File: boilerplate/app/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  function $textStyle({ pressed }) {
+    return [
+      $textPresets[preset],
+      $textStyleOverride,
+      !!pressed && [$pressedTextPresets[preset], $pressedTextStyleOverride],
+    ]
+  }
+
+  return (
+    <Pressable style={$viewStyle} {...rest}>
```
Comment (yulolimum) [inline boilerplate/app/components/Button.tsx:100]: Button is now a Pressable component which supports press states making it more customizable.

File: boilerplate/app/components/Screen.tsx [hunk 6]
```diff
@@ -0,0 +1,243 @@
+import React, { useRef, useState } from "react"
```
Comment (yulolimum) [inline boilerplate/app/components/Screen.tsx:1]: Screen component has been mostly DRYed up and made more customizable by exposing each sub-component props.

File: boilerplate/app/components/Screen.tsx [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+interface AutoScreenProps extends Omit<ScrollScreenProps, "preset"> {
+  preset?: "auto"
+  /**
+   * Threshold to trigger the automatic disabling/enabling of scroll ability.
+   * Defaults to `{ percent: 0.92 }`.
+   */
+  scrollEnabledToggleThreshold?: { percent?: number; point?: number }
+}
+
+export type ScreenProps = ScrollScreenProps | FixedScreenProps | AutoScreenProps
```
Comment (yulolimum) [inline boilerplate/app/components/Screen.tsx:86]: Makes the various screen types a discriminating union using the `preset` prop. This allows for better vscode suggestions for the prop differences.

File: boilerplate/app/components/Screen.tsx [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+const isIos = Platform.OS === "ios"
+
+function isNonScrolling(preset?: ScreenProps["preset"]) {
+  return !preset || preset === "fixed"
+}
+
+function useAutoPreset(props: AutoScreenProps) {
```
Comment (yulolimum) [inline boilerplate/app/components/Screen.tsx:94]: The alternative `auto` preset's functionality has been refactored into a hook to clean up the `scroll` preset component.

File: boilerplate/app/components/Screen.tsx [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+
+function ScreenWithoutScrolling(props: ScreenProps) {
+  const { style, contentContainerStyle, children } = props
+  return (
+    <View style={[$outerStyle, style]}>
+      <View style={[$innerStyle, contentContainerStyle]}>{children}</View>
+    </View>
+  )
+}
+
```
Comment (yulolimum) [inline boilerplate/app/components/Screen.tsx:152]: Cleans up both `ScreenWithoutScrolling` and `ScreenWithScrolling` components. They used to have a lot of duplicated code.

File: boilerplate/app/components/Text.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
+import i18n from "i18n-js"
```
Comment (yulolimum) [inline boilerplate/app/components/Text.tsx:1]: This is the consolidated Text component employing the proposed stylistic changes. The component is a lot more customizable now with the addition of `size` and `weight` props.

File: boilerplate/app/components/bullet-item/bullet-item.tsx [hunk 11]
```diff
@@ -21,7 +21,7 @@ const BULLET: ImageStyle = {
 }
 const BULLET_TEXT: TextStyle = {
   flex: 1,
-  fontFamily: typography.primary,
+  fontFamily: typography.primary.normal,
```
Comment (yulolimum) [inline boilerplate/app/components/bullet-item/bullet-item.tsx:24]: Changes to files below this one when it comes to colors and fonts are all temporary fixes and may completely break the look of the current (now legacy) components. These will all be addressed as more components are rebuilt.

File: boilerplate/app/components/index.ts [hunk 12]
```diff
@@ -1,13 +1,13 @@
+export * from "./Text"
+export * from "./Screen"
+export * from "./Button"
```
Comment (yulolimum) [inline boilerplate/app/components/index.ts:3]: Boom

File: boilerplate/app/screens/WelcomeScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,81 @@
+import { StackScreenProps } from "@react-navigation/stack"
```
Comment (yulolimum) [inline boilerplate/app/screens/WelcomeScreen.tsx:1]: This is the updated Welcome screen.

File: boilerplate/app/theme/colors.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
+// MAVERICKTODO: write documentation for colors and palette in own markdown file and add links from here
+
```
Comment (yulolimum) [inline boilerplate/app/theme/colors.ts:2]: Consolidates palette and colors into single file.

File: boilerplate/app/theme/typography.ts [hunk 15]
```diff
@@ -1,31 +1,71 @@
+// MAVERICKTODO: write documentation about fonts and typography along with guides on how to add custom fonts in own
+// markdown file and add links from here
+
```
Comment (yulolimum) [inline boilerplate/app/theme/typography.ts:3]: consolidates fonts and typography into single file.

File: boilerplate/storybook/index.ts [hunk 16]
```diff
@@ -1,4 +0,0 @@
-// this is the native storybook entry point
```
Comment (yulolimum) [inline boilerplate/storybook/index.ts:1]: goodbye

File: boilerplate/app/theme/colors.ts (hunk 14 above)
Comment (mazenchami) [reply boilerplate/app/theme/colors.ts:2]: i link it when its consolidated

Comment (jamonholmgren) [reply boilerplate/app/theme/colors.ts:2]: Yeah, this is nice!

**Review by jamonholmgren — APPROVED** [review]
Everything looks good (esp for a WIP). Very nice work on this Yulian. I love the direciton we're going with Ignite Maverick.

**Pull request merged**